pkg/flowaggregator/flowaggregator.go

```diff
@@ -321,46 +337,229 @@ func (fa *flowAggregator) initExportingProcess() error {
 }
 
 func (fa *flowAggregator) Run(stopCh <-chan struct{}) {
-	exportTicker := time.NewTicker(fa.exportInterval)
-	defer exportTicker.Stop()
 	go fa.collectingProcess.Start()
 	defer fa.collectingProcess.Stop()
 	go fa.aggregationProcess.Start()
 	defer fa.aggregationProcess.Stop()
+	go fa.flowRecordExpiryCheck(true, stopCh)
+	go fa.flowRecordExpiryCheck(false, stopCh)
+
+	<-stopCh
+}
+
+func (fa *flowAggregator) flowRecordExpiryCheck(isActiveTimeout bool, stopCh <-chan struct{}) {
+	var timeout time.Duration
+	var updatePQ ipfixintermediate.FlowKeyRecordMapCallBack
+	if isActiveTimeout {
+		timeout = fa.activeFlowRecordTimeout
+		updatePQ = fa.updateActiveExpiryPriorityQueue
+	} else {
+		timeout = fa.inactiveFlowRecordTimeout
+		updatePQ = fa.updateInactiveExpiryPriorityQueue
+	}
+
+	expireTicker := time.NewTicker(timeout)
 	for {
 		select {
 		case <-stopCh:
 			if fa.exportingProcess != nil {
 				fa.exportingProcess.CloseConnToCollector()
 			}
+			expireTicker.Stop()
 			return
-		case <-exportTicker.C:
+		case <-expireTicker.C:
 			if fa.exportingProcess == nil {
 				err := fa.initExportingProcess()
 				if err != nil {
-					klog.Errorf("Error when initializing exporting process: %v, will retry in %s", err, fa.exportInterval)
-					// Initializing exporting process fails, will retry in next exportInterval
+					klog.Errorf("Error when initializing exporting process: %v, will retry in %s", err, fa.activeFlowRecordTimeout)
```

Comment:
I think we can directly put `timeout` in error message so that if something went wrong, we can know whether it is in active cycle or inactive cycle.